isofit/utils/reducers.py

```diff
@@ -0,0 +1,63 @@
+#! /usr/bin/env python3
+#
+#  Copyright 2019 California Institute of Technology
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# ISOFIT: Imaging Spectrometer Optimal FITting
+# Author: David R Thompson, david.r.thompson@jpl.nasa.gov
+
+import numpy as np
+
+
+def band_mean(vals):
+    """
+    Wrapper for the np.mean reducer. A wrapper is necessary
+    because you need to pass it a row-axis call.
+
+    Args:
+        vals: (N-locs x bands) array of pixel values
+
+    Returns:
+        (b x 1) array of band-means
+    """
+    return np.mean(vals, axis=0)
+
+
+def class_priority(vals, thresh=0.25):
```

Comment:
An example of one strategy to reduce the full spatial resolution class assignment to super a pixel class assignment. Uses a hierarchy of surface types and a threshold percentage to be used in the hierarchy. 

The implementation is general enough s.t. you can replace with any function.